src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp

```diff
@@ -166,10 +166,9 @@ void TaskBooleanParameters::onSelectionChanged(const Gui::SelectionChanges& msg)
 
                 QString internalName = QString::fromStdString(body);
                 for (int row = 0; row < ui->listWidgetBodies->count(); row++) {
-                    QListWidgetItem* item = ui->listWidgetBodies->item(row);
+                    QListWidgetItem* item = ui->listWidgetBodies->takeItem(row);
                     QString name = item->data(Qt::UserRole).toString();
                     if (name == internalName) {
-                        ui->listWidgetBodies->takeItem(row);

```

Comment:
If my comment above is correct, I think this line will still be needed. Otherwise the `ui` `QListWidget` will be holding a freed pointer after the `delete item;` call (and still showing the item in the UI iiuc).

Reply:
No, it's possible to just delete an item without explicitly calling takeItem() beforehand but still using it is no harm and has almost no extra cost.

Reply:
@wwmayer I am very new to Qt (and FreeCAD for that matter), but it would seem without the `takeItem` call then the list would still contain a reference (pointer) to the (now freed) item as the element in position `row`. Is there something I am missing, does the destructor for the `item` contain a reference to the `ui` to remove itself from the list?

Looking at the [docs](https://doc.qt.io/qt-5/qlistwidget.html) I see:
> To remove items from the list, use takeItem().

fwiw: whether the `takeItem` call occurs before or after the `delete` isn't something I am concerned with. I am aware that you kept the call in the merged version, just want to make sure I understand.

Reply:
You can have a look at the Qt source code on-line and when deleting an item it also removes it from the item model.

> To remove items from the list, use takeItem().

Removing an item is not the same as deleting it. The takeItem() function is quite handy when you e.g. want to reorder a list view. So, you take it out from the view and add it back to another position.

Here is a simple example to show that directly deleting an item doesn't cause a crash:
```
    QListWidget* w = new QListWidget();
    w->show();

    w->addItem(new QListWidgetItem(QString::fromLatin1("1")));
    w->addItem(new QListWidgetItem(QString::fromLatin1("2")));
    w->addItem(new QListWidgetItem(QString::fromLatin1("3")));

    QListWidgetItem* item = w->item(1);
    delete item;
```